app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -357,6 +364,27 @@ def remove_network_adapter_config_spec(vim_obj, vmcs, hardware, options)
     end
   end
 
+  def edit_network_adapter_config_spec(vim_obj, vmcs, hardware, options)
+    network_name = options[:name]
+    network_vlan = options[:network]
+
+    _log.info("Change network #{network_name} to portGroup #{network_vlan}")
+    device = vim_obj.getDeviceByLabel(network_name, hardware)
+    raise "edit_network_adapter_config_spec: Network device with name #{network_name} was not found." if device.nil?
+
+    lan = Lan.find_by(:name => network_vlan, :switch_id => HostSwitch.where(:host_id => host).pluck(:switch_id))
```

Comment:
You can replace the `:switch_id => where().pluck()` with this: `Lan.find_by(:name => network_vlan, :switch => host.switches)`

Reply:
I will try this